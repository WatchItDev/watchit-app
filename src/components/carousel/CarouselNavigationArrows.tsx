import { FC } from 'react';
import { IconChevronLeft, IconChevronRight } from '@tabler/icons-react';

import IconButton from '@mui/material/IconButton';

import {NavigationArrowsProps} from "@src/components/carousel/types";

const CarouselNavigationArrows: FC<NavigationArrowsProps> = ({ next, prev }) => (
  <>
    <IconButton onClick={prev}>
      <IconChevronLeft />
    </IconButton>
    <IconButton onClick={next}>
      <IconChevronRight />
    </IconButton>
  </>
);

export default CarouselNavigationArrows;
