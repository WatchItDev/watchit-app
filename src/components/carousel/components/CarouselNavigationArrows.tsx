import { FC } from 'react'
import { IconChevronLeft, IconChevronRight } from '@tabler/icons-react'
import IconButton from '@mui/material/IconButton'
import { NavigationArrowsProps } from '@src/components/carousel/types.ts'

const CarouselNavigationArrows: FC<NavigationArrowsProps> = ({ next, prev }) => (
  <>
    <IconButton data-testid={'leftButton'} onClick={prev}>
      <IconChevronLeft />
    </IconButton>
    <IconButton data-testid={'rightButton'} onClick={next}>
      <IconChevronRight />
    </IconButton>
  </>
)

export default CarouselNavigationArrows
