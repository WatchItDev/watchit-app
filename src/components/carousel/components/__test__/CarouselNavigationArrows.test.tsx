import { render, fireEvent } from '@testing-library/react'
import { describe, it, expect, vi } from 'vitest'

import CarouselNavigationArrows from '@src/components/carousel/components/CarouselNavigationArrows'

describe('[COMPONENTS]: CarouselNavigationArrows component testing', () => {
  const prevMock = vi.fn()

  it('renders correctly with default props', () => {
    const { baseElement } = render(<CarouselNavigationArrows next={() => {}} prev={() => {}} />)
    expect(baseElement).toMatchSnapshot()
  })

  it('calls prev function when left arrow is clicked', () => {
    const { getByTestId} = render(<CarouselNavigationArrows next={() => {}} prev={prevMock} />)
    const leftArrow = getByTestId('leftButton')
    fireEvent.click(leftArrow)
    expect(prevMock).toHaveBeenCalledTimes(1)
  })

  it('calls next function when right arrow is clicked', () => {
    const nextMock = vi.fn()
    const { getByTestId } = render(<CarouselNavigationArrows next={nextMock} prev={() => {}} />)
    const rightArrow = getByTestId('rightButton')
    fireEvent.click(rightArrow)
    expect(nextMock).toHaveBeenCalledTimes(1)
  })

  it('renders left arrow icon correctly', () => {
    const { getByTestId } = render(<CarouselNavigationArrows next={() => {}} prev={() => {}} />)
    const leftArrow = getByTestId('leftButton')
    expect(leftArrow.querySelector('svg')).toBeInTheDocument()
  })

  it('renders right arrow icon correctly', () => {
    const { getByTestId } = render(<CarouselNavigationArrows next={() => {}} prev={() => {}} />)
    const rightArrow = getByTestId('rightButton')
    expect(rightArrow.querySelector('svg')).toBeInTheDocument()
  })
})
