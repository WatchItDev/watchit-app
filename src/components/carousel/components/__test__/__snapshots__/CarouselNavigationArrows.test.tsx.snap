// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[COMPONENTS]: CarouselNavigationArrows component testing > renders correctly with default props 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      data-testid="leftButton"
      tabindex="0"
      type="button"
    >
      <svg
        class="tabler-icon tabler-icon-chevron-left "
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15 6l-6 6l6 6"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      data-testid="rightButton"
      tabindex="0"
      type="button"
    >
      <svg
        class="tabler-icon tabler-icon-chevron-right "
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 6l6 6l-6 6"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;
